### PR TITLE
camera config

### LINKED ISSUE #183: undefined symbol error with openni2_launch
Occasionally I see the following error when launching `rosrun openni2_launch openni2.launch` after source `use_ros` and `use_spartan`.

If you only call `use_ros` then it works fine. My guess is that this is due to spartan building it's own libopenni2.

```
ROS_MASTER_URI=http://localhost:11311
]2;/home/manuelli/spartan/build/install/share/camera_config/launch/openni2.launch http://localhost:11311
core service [/rosout] found
process[camera_1112170110/camera_1112170110_nodelet_manager-1]: started with pid [31178]
[ INFO] [1509732042.987475247]: Initializing nodelet with 4 worker threads.
process[camera_1112170110/driver-2]: started with pid [31179]
/opt/ros/kinetic/lib/nodelet/nodelet: symbol lookup error: /opt/ros/kinetic/lib//libopenni2_camera_nodelet.so: undefined symbol: _ZN15openni2_wrapper13OpenNI2DriverC1ERN3ros10NodeHandleES3_
[FATAL] [1509732043.057331057]: Failed to load nodelet '/camera_1112170110/driver` of type `openni2_camera/OpenNI2DriverNodelet` to manager `camera_1112170110_nodelet_manager'
process[camera_1112170110/rgb_rectify_color-3]: started with pid [31200]
[camera_1112170110/camera_1112170110_nodelet_manager-1] process has died [pid 31178, exit code 127, cmd /opt/ros/kinetic/lib/nodelet/nodelet manager __name:=camera_1112170110_nodelet_manager __log:=/home/manuelli/.ros/log/88e220a2-c0be-11e7-83f3-0242ac110002/camera_1112170110-camera_1112170110_nodelet_manager-1.log].
log file: /home/manuelli/.ros/log/88e220a2-c0be-11e7-83f3-0242ac110002/camera_1112170110-camera_1112170110_nodelet_manager-1*.log
process[camera_1112170110/depth_rectify_depth-4]: started with pid [31216]
process[camera_1112170110/depth_metric_rect-5]: started with pid [31233]
process[camera_1112170110/depth_metric-6]: started with pid [31252]
[camera_1112170110/driver-2] process has died [pid 31179, exit code 255, cmd /opt/ros/kinetic/lib/nodelet/nodelet load openni2_camera/OpenNI2DriverNodelet camera_1112170110_nodelet_manager ir:=ir rgb:=rgb depth:=depth depth_registered:=depth_registered rgb/image:=rgb/image_raw depth/image:=depth_registered/image_raw __name:=driver __log:=/home/manuelli/.ros/log/88e220a2-c0be-11e7-83f3-0242ac110002/camera_1112170110-driver-2.log].
log file: /home/manuelli/.ros/log/88e220a2-c0be-11e7-83f3-0242ac110002/camera_1112170110-driver-2*.log
process[camera_1112170110/depth_points-7]: started with pid [31276]
```

### PR DESCRIPTION
**Pull request by manuelli**
- adds calibration files for xtion. Both intrinsics and extrinsics
- publishes camera_frame to robot transform
- updates rviz config.
- set USE_KINECT to OFF in director superbuild to address #183 

Fixes #183 

Ready for merge

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/spartan/181)

### CODE REVIEW

**Comment by gizatt** [comment]
Lots of stuff, so I don't think I can do a detailed review (it'll be more useful from my perspective to get it in and generally try it out).

I'm seeing a CMake error on Jenkins when configuring Director:

```
14:52:40 Scanning dependencies of target vtkPCLFilters
14:52:40 CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
14:52:40 Please set them or make sure they are set and tested correctly in the CMake files:
14:52:40 CVUTILS_INCLUDE_DIR
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40    used as include directory in directory /home/jenkins/spartan/director/src/app
14:52:40 CVUTILS_LIBRARY
14:52:40     linked by target "ddApp" in directory /home/jenkins/spartan/director/src/app
14:52:40 
```

You got any leads?

---

Reviewed 25 of 25 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/catkin_projects/camera_config/src/camera_config/camera_info_publisher.py, line 72 at r1](https://reviewable.io:443/reviews/robotlocomotion/spartan/181#-Ky2aavbcKFxOnD_0mM-:-Ky2aavbcKFxOnD_0mM0:byio0ri) ([raw file](https://github.com/robotlocomotion/spartan/blob/3b12ffbcef06fcb69eb4faf5c526b0fe29e65105/src/catkin_projects/camera_config/src/camera_config/camera_info_publisher.py#L72)):*
> ```Python
> 			print t
> 			rospy.loginfo("broadcasting frame for camera %s", key)
> 			# self.staticTransformBroadcaster.sendTransform(t)
> ```

Should this still be commented out?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/181#-:-Ky2amHEaBsTnyQcAMcs:bstyskv)*

**Comment by manuelli** [comment]
Yeah I set USE_KINNECT to OFF in director due to the libopenni2 issue. I didn't check the build after that. cvutils was part of that kinect stuff, so maybe we aren't allowed to have pcl filters and the kinect stuff off. I can dig into it.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/catkin_projects/camera_config/src/camera_config/camera_info_publisher.py, line 72 at r1](https://reviewable.io:443/reviews/robotlocomotion/spartan/181#-Ky2aavbcKFxOnD_0mM-:-Ky2gD4P7Wbf0e0Jk0Vj:b-5e3ez9) ([raw file](https://github.com/robotlocomotion/spartan/blob/3b12ffbcef06fcb69eb4faf5c526b0fe29e65105/src/catkin_projects/camera_config/src/camera_config/camera_info_publisher.py#L72)):*
<details><summary><i>Previously, gizatt (Gregory Izatt) wrote…</i></summary><blockquote>

Should this still be commented out?
</blockquote></details>

I should remove that. I send the transform with the standard broadcaster 3 lines above.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/181#-:-Ky2gMCFw6e8pwAV8IQW:b-d0d8n7)*

**Comment by gizatt** [comment]
jenkins please test

**Comment by gizatt** [comment]
:LGTM:

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/181#-:-Ky2zIVMUA6qD0TvySNU:bmxilld)*

File: apps/iiwa/iiwa_hardware.pmd [hunk 1]
```diff
@@ -135,7 +135,12 @@ group "8.ROS" {
   }
 
   cmd "7.ROS-OpenNI" {
-    exec = "roslaunch openni2_launch openni2.launch";
+    exec = "roslaunch camera_config openni2.launch";
+    host = "localhost";
+  }
+
+  cmd "8.camera_info_publisher" {
```
Comment (peteflorence) [inline apps/iiwa/iiwa_hardware.pmd]: camera transform publisher?

**Comment by peteflorence** [comment]
LGTM after camera transform publisher name switch

**Comment by manuelli** [comment]
Changes have been made

---

Review status: 19 of 26 files reviewed at latest revision, 2 unresolved discussions.

---

*[apps/iiwa/iiwa_hardware.pmd, line 142 at r2](https://reviewable.io:443/reviews/robotlocomotion/spartan/181#-KxyzcmJlLoxSAOec2Z4-r2-142:-Ky3TYoxExcCgGojKB-c:b-896fix) ([raw file](https://github.com/robotlocomotion/spartan/blob/46738e643acdc7c0a0671277063d8b9b1ec28dac/apps/iiwa/iiwa_hardware.pmd#L142)):*
<details><summary><i>Previously, peteflorence (Pete Florence) wrote…</i></summary><blockquote>

camera transform publisher?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/181#-:-Ky3TAAm59T06ztIkAWz:b-girykt)*

**Pull request merged**